app.py

```diff
@@ -0,0 +1,244 @@
+        State("radio", "value"),
+    ],
+    running=[
+        # Shows cancel button while running.
+        (Output("cancel-button", "className"), "", "display-none"),
+        (Output("run-button", "className"), "display-none", ""),  # Hides run button while running.
+        (Output("results-tab", "disabled"), True, False),  # Disables results tab while running.
+        (Output("results-tab", "label"), "Loading...", "Results"),
+        (Output("tabs", "value"), "input-tab", "input-tab"),  # Switch to input tab while running.
+        (Output("run-in-progress", "data"), True, False),  # Can block certain callbacks.
+    ],
+    cancel=[Input("cancel-button", "n_clicks")],
+    prevent_initial_call=True,
+)
+def run_optimization(
+    # The parameters below must match the `Input` and `State` variables found
+    # in the `inputs` list above.
+    run_click: int,
+    sampler_type: Union[SamplerType, int],
+    time_limit: float,
+    slider_value: int,
+    dropdown_value: int,
+    checklist_value: list,
+    radio_value: int,
+) -> RunOptimizationReturn:
+    """Runs the optimization and updates UI accordingly.
+
+    This is the main function which is called when the `Run Optimization` button is clicked.
+    This function takes in all form values and runs the optimization, updates the run/cancel
+    buttons, deactivates (and reactivates) the results tab, and updates all relevant HTML
+    components.
+
+    Args:
+        run_click: The (total) number of times the run button has been clicked.
+        sampler_type: Either Quantum Hybrid (``0`` or ``SamplerType.HYBRID``),
+            or Classical (``1`` or ``SamplerType.CLASSICAL``).
+        time_limit: The solver time limit.
+        slider_value: The value of the slider.
+        dropdown_value: The value of the dropdown.
+        checklist_value: A list of the values of the checklist.
+        radio_value: The value of the radio.
+
+    Returns:
+        A NamedTuple (RunOptimizationReturn) containing all outputs to be used when updating the HTML
+        template (in ``dash_html.py``). These are:
+
+            results: The results to display in the results tab.
+            problem-details: List of the table rows for the problem details table.
+    """
+
+    # Only run optimization code if this function was triggered by a click on `run-button`.
+    # Setting `Input` as exclusively `run-button` and setting `prevent_initial_call=True`
+    # also accomplishes this.
+    if run_click == 0 or ctx.triggered_id != "run-button":
+        raise PreventUpdate
+
+    if isinstance(sampler_type, int):
+        sampler_type = SamplerType(sampler_type)
+
+    print(
+        f"The form has the following values:\n\
+        Example Slider: {slider_value}\n\
+        Example Dropdown: {dropdown_value}\n\
+        Example Checklist: {checklist_value}\n\
+        Example Radio: {radio_value}\n\
+        Solver: {sampler_type}\n\
+        Time Limit: {time_limit}"
+    )
+
```

Comment:
Maybe remove this since it's yet another section to update/remove for the user.
```suggestion
```

Reply:
I put it here so the user can get feedback about the settings without having to change anything. I don't expect them to keep it just to use it for discovery purposes. I can add a comment to explain that

Reply:
Added a comment to explain this, if you think it's too complicated let me know and I'll just axe the whole thing https://github.com/dwave-examples/template-dash/pull/2/commits/c455082d70a480fac66fc4c758629f971bf4bda1

Reply:
I still think it's a bit unnecessary, but this is not a very strong opinion if you'd prefer to keep it.

Reply:
Sounds good, I've removed it https://github.com/dwave-examples/template-dash/pull/2/commits/9c790f0ea295d743bf94ecbdeec9ee1e7b21a473